src/twisted/web/tap.py

```diff
@@ -194,18 +213,40 @@ def postOptions(self):
         If no server port was supplied, select a default appropriate for the
         other options supplied.
         """
+        if self['port']:
+            def opt_port():
```

Comment:
Why not make this an instance method and then deprecate that?

Reply:
It's a lot of stuff to do it, but sure.